clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -449,9 +472,28 @@ scheduler_impl_tmpl_t<trace_entry_t, record_reader_t>::record_type_set_pid(
 template <>
 bool
 scheduler_impl_tmpl_t<trace_entry_t, record_reader_t>::record_type_is_instr(
-    trace_entry_t record)
+    trace_entry_t record, addr_t *pc, size_t *size)
 {
-    return type_is_instr(static_cast<trace_type_t>(record.type));
+    if (type_is_instr(static_cast<trace_type_t>(record.type))) {
+        if (pc != nullptr)
+            *pc = record.addr;
+        if (size != nullptr)
+            *size = record.size;
+        return true;
+    }
+    return false;
+}
+
+template <>
+bool
+scheduler_impl_tmpl_t<trace_entry_t, record_reader_t>::
+    record_type_is_indirect_branch_instr(trace_entry_t &record,
+                                         addr_t set_indirect_branch_target_unused)
+{
+    // Cannot set the provided indirect branch target here because
+    // a prior trace_entry_t would have it.
```

Comment:
Should it be an error if non-0 is passed then?  The caller shouldn't pass non-0 and expect that to add a marker record.

Reply:
We still want this API to let the caller know whether the record is an indirect branch (that is its primary purpose). The API documentation notes that the indirect_branch_target is set only for memref_t. We want to avoid having to specialize the caller for memref_t and trace_entry_t.

Reply:
Modified the api to return has_indirect_branch_target which will be false for trace_entry_t which additionally lets the user know that the field wasn't set.